src/components/TextInput/TextInput.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import colors from '../../theme/colors.scss';
+
+import './TextInput.scss';
+
+const prefixCls = 'kai-text-input';
+
+class TextInput extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isFocused: false,
+    };
+    this.handleFocusChange = this.handleFocusChange.bind(this);
+  }
+
+  handleFocusChange(isFocused) {
+    this.setState({ isFocused });
+    if (isFocused) {
+      this.props.onFocusChange(this.props.index);
+      this.props.forwardedRef.current.focus();
+    }
+  }
+
+  render() {
+    const { focusColor, forwardedRef, label, onChange } = this.props;
+    const { isFocused } = this.state;
+
+    const itemCls = classnames([
+      prefixCls,
+      isFocused && `${prefixCls}--focused`,
+    ]);
+    const labelCls = `${prefixCls}-label`;
+    const inputCls = `${prefixCls}-input`;
+
+    return (
+      <div
+        tabIndex="0"
+        className={itemCls}
+        style={{ backgroundColor: isFocused ? focusColor : colors.white }}
+        onFocus={() => this.handleFocusChange(true)}
+        onBlur={() => this.handleFocusChange(false)}
+      >
+        <label className={labelCls}>{label}</label>
+        <input
+          ref={forwardedRef}
+          type="text"
+          className={inputCls}
+          onChange={onChange}
+        />
+      </div>
+    );
+  }
+}
+
+TextInput.defaultProps = {
+  focusColor: colors.defaultFocusColor,
+  onChange: () => {},
+};
+
+TextInput.propTypes = {
```

Comment:
I am not sure if we need some more props here so we can set attributes like autocomplete, autofocus, disabled, pattern and placeholder directly in the component instead on changing the input element itself.

Reply:
Ohhh, that would be cool. Maybe not necessary for MVP (don't want to block Daniel) but we can add it to the roadmap? This is applicable to other components as well.

Reply:
I figured a good approach to this would be to just allow passing down of any props that aren't explicitly defined to the input field. That way we don't have to have an exhaustive list of properties that should be supported.